### PR TITLE
FSET 1641: Numeric Test Extract Report

### PR DESCRIPTION
**Pull request by henricook**
Show users who are in Sift (but not completed) and require a manual invite to numeric tests by the service team

### CODE REVIEW

File: app/controllers/ReportingController.scala [hunk 1]
```diff
@@ -272,4 +273,27 @@ trait ReportingController extends BaseController {
       Ok(Json.toJson(list))
     }
   }
+
+  def numericTestExtractReport(): Action[AnyContent] = Action.async { implicit request =>
+
+    val numericTestSchemeIds = schemeRepo.schemes.filter(scheme => scheme.siftEvaluationRequired &&
```
Comment (benjstephenson) [inline app/controllers/ReportingController.scala]: y u no collect?

File: app/controllers/ReportingController.scala [hunk 2]
```diff
@@ -272,4 +273,27 @@ trait ReportingController extends BaseController {
...
+      scheme.siftRequirement.contains(SiftRequirement.NUMERIC_TEST)).map(_.id)
+
+    val reports =
+      for {
+        applications <- reportingRepository.numericTestExtractReport().map(_.filter { app =>
+          val successfulSchemesSoFarIds = app.currentSchemeStatus.filter(_.result == Green.toString).map(_.schemeId)
+          successfulSchemesSoFarIds.exists(numericTestSchemeIds.contains)
+        })
+        contactDetails <- contactDetailsRepository.findByUserIds(applications.map(_.userId)).map(_.map(x => x.userId -> x)(breakOut).toMap)
```
Comment (benjstephenson) [inline app/controllers/ReportingController.scala]: specifying the type on the val assignment would remove the need for the explicit toMap call

File: app/model/persisted/ApplicationForNumericTestExtractReport.scala [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+package model.persisted
+
+import model.ApplicationRoute._
+import model.SchemeId
+import play.api.libs.json.Json
+import model.report.TestResultsForOnlineTestPassMarkReportItem
+import reactivemongo.bson.Macros
+
+case class ApplicationForNumericTestExtractReport(
```
Comment (benjstephenson) [inline app/model/persisted/ApplicationForNumericTestExtractReport.scala]: pedantic pants - formatting

File: app/controllers/ReportingController.scala (hunk 2 above)
Comment (henricook) [inline app/controllers/ReportingController.scala]: Yar but it's more characters :-D

File: app/model/persisted/ApplicationForNumericTestExtractReport.scala (hunk 3 above)
Comment (henricook) [inline app/model/persisted/ApplicationForNumericTestExtractReport.scala]: One of our many tools did this on its own, i've reformatted with intellij

File: app/repositories/application/ReportingRepository.scala [hunk 4]
```diff
@@ -45,7 +45,9 @@ trait ReportingRepository {
 
   def diversityReport(frameworkId: String): Future[List[ApplicationForDiversityReport]]
 
-  def onlineTestPassMarkReport(frameworkId: String): Future[List[ApplicationForOnlineTestPassMarkReport]]
+  def onlineTestPassMarkReport(): Future[List[ApplicationForOnlineTestPassMarkReport]]
```
Comment (benjstephenson) [inline app/repositories/application/ReportingRepository.scala]: Do you need these parentheses?  No side effects here that I can see?

File: test/connectors/CubiksGatewayClientSpec.scala [hunk 5]
```diff
@@ -130,7 +130,7 @@ class CubiksGatewayClientSpec extends UnitSpec with ShortTimeout {
 
     def mockPost[T] = {
       when(
-        mockWSHttp.POST(anyString(), any[T], anyObject())(any[Writes[T]], any[HttpReads[HttpResponse]], any[HeaderCarrier])
+        mockWSHttp.POST(anyString(), any[T], any())(any[Writes[T]], any[HttpReads[HttpResponse]], any[HeaderCarrier])
```
Comment (benjstephenson) [inline test/connectors/CubiksGatewayClientSpec.scala]: nice

**Pull request merged**